akd_core/src/proto/tests.rs

```diff
@@ -278,3 +278,33 @@ fn test_convert_single_append_only_proof() {
     let protobuf: SingleAppendOnlyProof = (&original).into();
     assert_eq!(original, (&protobuf).try_into().unwrap());
 }
+
+#[test]
+fn test_minimum_encoding_label_bytes() {
```

Comment:
Tests might just set an actual NodeLabel with label_val and label_len and make sure it is truncated properly

Reply:
Spoke out-of-band, since label's might allow for additional 0's added, this optimizes for wire-size above all else (nodes in the very left of the tree will have trailing 0's, and the length will point beyond the last sig byte). This "search" is a small computation to minimize the wire-size overall.

Tests with node-label will have no effect then, since we don't utilize the length anyways.